### PR TITLE
ANDAUTH-43

### PR DESCRIPTION
**Pull request by jaindiv26**
Added test cases for ServerID, SignIn, SignUp, groups loading.

### CODE REVIEW

File: app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.xwiki.android.sync.activities
+
+import org.junit.After
+import org.junit.Before
+
+/**
+ * EditContactActivityTest
+ */
+class EditContactActivityTest {
```
Comment (InsanusMokrassar) [inline app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt]: Why have you kept this empty test?

File: app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt [hunk 2]
```diff
@@ -68,8 +69,15 @@ class SignInViewFlipper(activity: AuthenticatorActivity, contentRootView: View)
      */
     private lateinit var accountPassword: String
 
+    /**
+     * Databinding of this flipper
+     */
+
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt:74]: Redundant line

**Review by InsanusMokrassar — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt (hunk 1 above)
Comment (jaindiv26) [reply app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt]: I've kept it coz, I'll be starting my work on EditContactActivity in coming weeks in which I'll be writing test cases.

File: app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt (hunk 2 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt:74]: Removed.

File: app/build.gradle [hunk 3]
```diff
@@ -25,6 +25,12 @@ android {
     dataBinding {
         enabled = true
     }
+    configurations {
+        zipped
```
Comment (InsanusMokrassar) [inline app/build.gradle]: Is it required?

File: app/src/androidTest/java/org/xwiki/android/sync/auth/AuthenticatorActivityTest.kt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        activityScenario.moveToState(Lifecycle.State.STARTED)
+        activityScenario.close()
+    }
+
+    @Test
+    fun testSignIn() {
+        activityScenario.onActivity {
+            it.showViewFlipper(0)
+        }
+//        onView(withId(R.id.accountServer)).perform(clearText()) // for testing on local host uncomment these line
```
Comment (InsanusMokrassar) [inline app/src/androidTest/java/org/xwiki/android/sync/auth/AuthenticatorActivityTest.kt]: Remove this comments

File: app/src/androidTest/java/org/xwiki/android/sync/contactdb/ContactManagerTest.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import androidx.test.runner.AndroidJUnit4
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * ContactManagerTest
+ */
+@RunWith(AndroidJUnit4::class)
+class ContactManagerTest {
```
Comment (InsanusMokrassar) [inline app/src/androidTest/java/org/xwiki/android/sync/contactdb/ContactManagerTest.kt]: What for?

File: app/src/androidTest/java/org/xwiki/android/sync/rest/HttpExecutorTest.kt [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package org.xwiki.android.sync.rest
+
+import org.junit.Before
+import org.junit.Test
+
+/**
+ * HttpExecutorTest.
+ */
+class HttpExecutorTest {
```
Comment (InsanusMokrassar) [inline app/src/androidTest/java/org/xwiki/android/sync/rest/HttpExecutorTest.kt]: what for?

File: app/src/androidTest/java/org/xwiki/android/sync/rest/XWikiHttpTest.kt [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package org.xwiki.android.sync.rest
+
+import org.junit.Before
+
+/**
+ * XWikiHttpTest
+ */
+class XWikiHttpTest {
```
Comment (InsanusMokrassar) [inline app/src/androidTest/java/org/xwiki/android/sync/rest/XWikiHttpTest.kt]: What for?

File: app/src/androidTest/java/org/xwiki/android/sync/utils/StringUtilAndroidTest.kt [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun iso8601ToDate() {
```
Comment (InsanusMokrassar) [inline app/src/androidTest/java/org/xwiki/android/sync/utils/StringUtilAndroidTest.kt:25]: Please, clear comments

File: app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt (hunk 1 above)
Comment (InsanusMokrassar) [inline app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt]: What for?

File: app/src/main/java/org/xwiki/android/sync/utils/EspressoIdlingResource.kt [hunk 9]
```diff
@@ -0,0 +1,18 @@
+package org.xwiki.android.sync.utils
+
+import androidx.test.espresso.IdlingResource
+
+    private val RESOURCE = "GLOBAL"
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/utils/EspressoIdlingResource.kt]: Please, fix tabulation

File: app/src/main/res/layout/viewflipper_setting_ip.xml [hunk 10]
```diff
@@ -52,6 +52,7 @@
                 />
         </com.google.android.material.textfield.TextInputLayout>
         <Button
+                android:id="@+id/btViewSignInFlipper"
```
Comment (InsanusMokrassar) [inline app/src/main/res/layout/viewflipper_setting_ip.xml:61]: Please, fix tabulation

File: app/src/test/java/org/xwiki/android/sync/utils/StringUtilsTest.kt [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    @Test
+    @Throws(Exception::class)
+    fun dateToIso8601String() {
+        assertNotNull(StringUtils.dateToIso8601String(Date()))
+        //System.out.println(StringUtils.dateToIso8601String(new Date()));
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun isValidServerAddress() {
```
Comment (InsanusMokrassar) [inline app/src/test/java/org/xwiki/android/sync/utils/StringUtilsTest.kt]: What this method testing?

File: authdemo/src/androidTest/java/org/xwiki/android/authdemo/ApplicationTest.java [hunk 12]
```diff
@@ -1,13 +1,10 @@
...
-import android.test.ApplicationTestCase;
 
 /**
  * <a href="http://d.android.com/tools/testing/testing_android.html">Testing Fundamentals</a>
  */
-public class ApplicationTest extends ApplicationTestCase<Application> {
-    public ApplicationTest() {
-        super(Application.class);
-    }
+public class ApplicationTest  {
```
Comment (InsanusMokrassar) [inline authdemo/src/androidTest/java/org/xwiki/android/authdemo/ApplicationTest.java]: What for?

**Review by InsanusMokrassar — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/xwiki/android/sync/contactdb/ContactManagerTest.kt (hunk 5 above)
Comment (jaindiv26) [reply app/src/androidTest/java/org/xwiki/android/sync/contactdb/ContactManagerTest.kt]: Deleted

File: app/src/androidTest/java/org/xwiki/android/sync/rest/HttpExecutorTest.kt (hunk 6 above)
Comment (jaindiv26) [reply app/src/androidTest/java/org/xwiki/android/sync/rest/HttpExecutorTest.kt]: Deleted

File: app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt (hunk 1 above)
Comment (jaindiv26) [reply app/src/androidTest/java/org/xwiki/android/sync/activities/EditContactActivityTest.kt]: Deleted

File: app/src/androidTest/java/org/xwiki/android/sync/auth/AuthenticatorActivityTest.kt (hunk 4 above)
Comment (jaindiv26) [reply app/src/androidTest/java/org/xwiki/android/sync/auth/AuthenticatorActivityTest.kt]: Done

File: app/build.gradle (hunk 3 above)
Comment (jaindiv26) [reply app/build.gradle]: Removed

File: app/src/androidTest/java/org/xwiki/android/sync/rest/XWikiHttpTest.kt (hunk 7 above)
Comment (jaindiv26) [reply app/src/androidTest/java/org/xwiki/android/sync/rest/XWikiHttpTest.kt]: Removed

File: app/src/androidTest/java/org/xwiki/android/sync/utils/StringUtilAndroidTest.kt (hunk 8 above)
Comment (jaindiv26) [reply app/src/androidTest/java/org/xwiki/android/sync/utils/StringUtilAndroidTest.kt:25]: uncommented lines.

File: app/src/main/java/org/xwiki/android/sync/utils/EspressoIdlingResource.kt (hunk 9 above)
Comment (jaindiv26) [reply app/src/main/java/org/xwiki/android/sync/utils/EspressoIdlingResource.kt]: Done

File: app/src/main/res/layout/viewflipper_setting_ip.xml (hunk 10 above)
Comment (jaindiv26) [reply app/src/main/res/layout/viewflipper_setting_ip.xml:61]: Done

File: authdemo/src/androidTest/java/org/xwiki/android/authdemo/ApplicationTest.java (hunk 12 above)
Comment (jaindiv26) [reply authdemo/src/androidTest/java/org/xwiki/android/authdemo/ApplicationTest.java]: Removed.

File: app/src/test/java/org/xwiki/android/sync/utils/StringUtilsTest.kt (hunk 11 above)
Comment (jaindiv26) [reply app/src/test/java/org/xwiki/android/sync/utils/StringUtilsTest.kt]: Removing them for now.

**Review by InsanusMokrassar — APPROVED** [review]

**Pull request merged**